tmp/lanelet2_extension/lib/mgrs_projector.cpp

```diff
@@ -0,0 +1,130 @@
+// Copyright 2015-2019 Autoware Foundation. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Authors: Simon Thompson, Ryohsuke Mitsudome
+
+// NOLINTBEGIN(readability-identifier-naming)
+
+#include "lanelet2_extension/projection/mgrs_projector.hpp"
+
+#include <iostream>
+#include <set>
+#include <string>
+#include <utility>
+#include <vector>
+
+namespace lanelet::projection
+{
+MGRSProjector::MGRSProjector(Origin origin) : Projector(origin) {}
+
+BasicPoint3d MGRSProjector::forward(const GPSPoint & gps) const
+{
+  BasicPoint3d mgrs_point(forward(gps, 0));
+  return mgrs_point;
+}
+
+BasicPoint3d MGRSProjector::forward(const GPSPoint & gps, const int precision) const
+{
+  std::string prev_projected_grid = projected_grid_;
+
+  BasicPoint3d mgrs_point{0., 0., gps.ele};
+  BasicPoint3d utm_point{0., 0., gps.ele};
+  int zone{};
+  bool northp{};
+  std::string mgrs_code;
+
+  try {
+    GeographicLib::UTMUPS::Forward(gps.lat, gps.lon, zone, northp, utm_point.x(), utm_point.y());
+    GeographicLib::MGRS::Forward(
+      zone, northp, utm_point.x(), utm_point.y(), gps.lat, precision, mgrs_code);
```

Comment:
https://geographiclib.sourceforge.io/2009-03/classGeographicLib_1_1UTMUPS.html

```suggestion
  bool is_north{};
  std::string mgrs_code;

  try {
    GeographicLib::UTMUPS::Forward(gps.lat, gps.lon, zone, is_north, utm_point.x(), utm_point.y());
    GeographicLib::MGRS::Forward(
      zone, is_north, utm_point.x(), utm_point.y(), gps.lat, precision, mgrs_code);
```